app/controllers/restaurants_controller.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+class RestaurantsController < ApplicationController
+  before_action :authenticate_user!
+
+  def index
+    @restaurants = Restaurant.all.includes(:orders)
+    new
```

Comment:
this looks really weird! Better to extract logic (building new restaurant) to before_action